src/api/common.ts

```diff
@@ -180,6 +181,11 @@ export interface WebsocketConnectionOptions {
    */
   url?: URL;
 
+  /**
+   * Options to pass to the underlying websocket connection.
+   */
+  options?: IsoWebSocket.ClientOptions;
```

Comment:
I'd slightly prefer that we don't expose the types we're using internally and provide a type we've defined here that contains just the options that we intend to expose. Just for the reason of being able to swap out the implementation with minimal interface changes. That's more of a style/maintainability thing though, this is just a type and isn't doing anything functional.

Reply:
Absolutely...so only the origin option for now?

Reply:
Nice yeah that looks good. Looks like the doc needs rebuilding?

Reply:
How would I do that? I ran npm run build before commiting which runs npm run build:docs as well...when I run it again locally nothing changes.

Reply:
Yeah, `npm run build` should do it. You are a contributor in this repo, so if you copy the branch to the repo, I'd give it a go too.

Reply:
I have no push access to the repo unfortunately, otherwise I would not have made a fork

Reply:
Can you try again? You should have push access.

Reply:
And sorry for not having done this earlier. I misread the permissions.

Reply:
Yep, just pushed

Reply:
Ah, I see now yes. Unfortunate that the name `WebsocketConnectionOptions` is already in use...maybe we want to rename this.

Reply:
Not sure the naming is optimal but the problem should be addressed with the latest commit.

Reply:
Yes, it is.
What do you think of this:
```typescript
/**
 * @public
 */
export type WsClientOptions = Pick<IsoWebSocket.ClientOptions, "origin">;

/**
 * Options for a Websocket connection.
 *
 * @public
 */
export interface WebsocketConnectionOptions {
  /**
   * The `ws://` URL of the Websocket server to connect to. Not required when connecting to App API from a Launcher or Kangaroo environment.
   */
  url?: URL;

  /**
   * Options to pass to the underlying websocket connection.
   */
  wsClientOptions?: WsClientOptions;
```

Reply:
Yes I think it's better - changed it.

Reply:
I also meant this
```typescript
/**
 * @public
 */
export type WsClientOptions = Pick<IsoWebSocket.ClientOptions, "origin">;
```

Reply:
I am confident that I will eventually get there ;). Gave it another try.

Reply:
So am I! =)

Thanks. I checked out the branch to look for the test errors and saw that all tests would need updating now, because `allowed_origin` becomes a mandatory param. That made me rethink this and I've pushed the suggestion to your branch in this repo. Please have a look at [the commit](https://github.com/holochain/holochain-client-js/commit/35131c333d68772c096babad45862f7e40b76b5e).

Reply:
I updated the existing tests with `allowed_origins` and added a test for this allowed_origins feature. It's failing at the moment because it's not been released on Holochain.